test/runtime_tests.cc

```diff
@@ -1898,5 +1898,62 @@ namespace detail {
 		CHECK(out == 43);
 	}
 
+	TEST_CASE("accessors deduces correctly from SYCL 2020 tags for device", "[accessor]") {
+		buffer<int, 1> buf_a(cl::sycl::range<1>(32));
+		auto& tm = runtime::get_instance().get_task_manager();
+
+		const auto tid = test_utils::add_compute_task<class get_access_with_tag>(
+		    tm,
+		    [&](handler& cgh) {
+			    accessor acc1{buf_a, cgh, one_to_one<1>(), cl::sycl::write_only};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::write, cl::sycl::target::device>, decltype(acc1)>);
+
+			    accessor acc2{buf_a, cgh, one_to_one<1>(), cl::sycl::read_only};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::read, cl::sycl::target::device>, decltype(acc2)>);
+
+			    accessor acc3{buf_a, cgh, one_to_one<1>(), cl::sycl::read_write};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::read_write, cl::sycl::target::device>, decltype(acc3)>);
+
+			    accessor acc4{buf_a, cgh, one_to_one<1>(), cl::sycl::write_only, cl::sycl::no_init};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::discard_write, cl::sycl::target::device>, decltype(acc4)>);
+
+			    accessor acc5{buf_a, cgh, one_to_one<1>(), cl::sycl::read_write, cl::sycl::no_init};
+			    static_assert(std::is_same_v<accessor<int, 1, cl::sycl::access_mode::discard_read_write, cl::sycl::target::device>, decltype(acc5)>);
+		    },
+		    buf_a.get_range());
+
+		const auto tsk = tm.get_task(tid);
+		REQUIRE(tsk->get_buffer_access_map().get_access_modes(0).count(cl::sycl::access_mode::write) == 1);
```

Comment:
Instead of `0`, consider using `detail::get_buffer_id` here to increase legibility

Reply:
done